src/main/java/org/spongepowered/common/mixin/core/entity/monster/MixinEntityEndermite.java

```diff
@@ -62,4 +70,15 @@ public void supplyVanillaManipulators(List<DataManipulator<?, ?>> manipulators)
         super.supplyVanillaManipulators(manipulators);
         manipulators.add(getExpirableData());
     }
+
+    @Inject(method = "onLivingUpdate",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/monster/EntityEndermite;setDead()V"))
+    private void fireExpireEventLifetime(CallbackInfo ci) {
+        try (CauseStackManager.StackFrame frame = Sponge.getCauseStackManager().pushCauseFrame()) {
+            Sponge.getCauseStackManager().pushCause(this);
```

Comment:
You can push this directly to the frame so you don't have to pop it manually.

Reply:
Ah, that makes sense. I'll update that accordingly